lib/backend/k8s/resources/networkpolicy.go

```diff
@@ -0,0 +1,295 @@
+		resource:        NetworkPolicyResourceName,
+		description:     "Calico Network Policies",
+		k8sResourceType: reflect.TypeOf(apiv2.NetworkPolicy{}),
+		k8sListType:     reflect.TypeOf(apiv2.NetworkPolicyList{}),
+		converter:       NetworkPolicyConverter{},
+		namespaced:      true,
+	}
+	return &networkPolicyClient{
+		clientSet: c,
+		crdClient: crdClient,
+		converter: conversion.Converter{},
+	}
+}
+
+// Implements the api.Client interface for NetworkPolicys.
+type networkPolicyClient struct {
+	clientSet *kubernetes.Clientset
+	crdClient *customK8sResourceClient
+	converter conversion.Converter
+}
+
+func (c *networkPolicyClient) Create(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
+	log.Debug("Received Create request on NetworkPolicy type")
+	key := kvp.Key.(model.ResourceKey)
+	if strings.HasPrefix(key.Name, "knp.default.") {
+		// We don't support Create of a Kubernetes NetworkPolicy.
+		return nil, cerrors.ErrorOperationNotSupported{
+			Identifier: kvp.Key,
+			Operation:  "Create",
+		}
+	}
+	return c.crdClient.Create(ctx, kvp)
+}
+
+func (c *networkPolicyClient) Update(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
+	log.Debug("Received Update request on NetworkPolicy type")
+	key := kvp.Key.(model.ResourceKey)
+	if strings.HasPrefix(key.Name, "knp.default.") {
+		// We don't support Update of a Kubernetes NetworkPolicy.
+		return nil, cerrors.ErrorOperationNotSupported{
+			Identifier: kvp.Key,
+			Operation:  "Update",
+		}
+	}
+	return c.crdClient.Update(ctx, kvp)
+}
+
+func (c *networkPolicyClient) Apply(kvp *model.KVPair) (*model.KVPair, error) {
+	return nil, cerrors.ErrorOperationNotSupported{
+		Identifier: kvp.Key,
+		Operation:  "Apply",
+	}
+}
+
+func (c *networkPolicyClient) Delete(ctx context.Context, key model.Key, revision string) (*model.KVPair, error) {
+	log.Debug("Received Delete request on NetworkPolicy type")
+	k := key.(model.ResourceKey)
+	if strings.HasPrefix(k.Name, "knp.default.") {
+		// We don't support Delete of a Kubernetes NetworkPolicy.
+		return nil, cerrors.ErrorOperationNotSupported{
+			Identifier: key,
+			Operation:  "Delete",
+		}
+	}
+	return c.crdClient.Delete(ctx, key, revision)
+}
+
+func (c *networkPolicyClient) Get(ctx context.Context, key model.Key, revision string) (*model.KVPair, error) {
+	log.Debug("Received Get request on NetworkPolicy type")
+	k := key.(model.ResourceKey)
+	if k.Name == "" {
+		return nil, errors.New("Missing policy name")
+	}
+
+	// Check to see if this is backed by a NetworkPolicy.
+	if strings.HasPrefix(k.Name, "knp.default.") {
+		// Backed by a NetworkPolicy. Parse out the namespace / name.
+		namespace, policyName, err := c.converter.ParsePolicyNameNetworkPolicy(k.Name)
+		if err != nil {
+			return nil, cerrors.ErrorResourceDoesNotExist{Err: err, Identifier: k}
+		}
+
+		// Get the NetworkPolicy from the API and convert it.
+		networkPolicy := extensions.NetworkPolicy{}
+		err = c.clientSet.Extensions().RESTClient().
```

Comment:
Shouldn't we be using the clientset.NetworkingV1() client rather than the rest client here?

Reply:
and elsewhere.

Reply:
not yet